tests/unit/optimizer_test.py

```diff
@@ -19,170 +19,175 @@
-
-    # Start right of local max pi/2
-    optimizedThreshold, optimizedScore = twiddle(objFunction=sine,
-      args = (),
-      initialGuess = math.pi*(0.5 + 0.1),
-      domain = (0, 2*math.pi))
-
-    self.assertTrue(abs(optimizedThreshold - math.pi/2) <= 0.00001,
-      "Optimizer returned x = %r, which is not within the tolerance of the \
-      maximum location x = %r" % (optimizedThreshold, math.pi/2))
-    self.assertTrue(abs(optimizedScore - 1.0) <= 0.00001,
-      "Optimizer returned max value of %r, but expected 1.0" % optimizedScore)
-
-    # Start left of local min
-    optimizedThreshold, optimizedScore = twiddle(objFunction=sine,
-      args = (),
-      initialGuess = math.pi*(1.5 - 0.1),
-      domain = (0, 2*math.pi))
-
-    self.assertTrue(abs(optimizedThreshold - math.pi/2) <= 0.00001,
-      "Optimizer returned x = %r, which is not within the tolerance of the \
-      maximum location x = %r" % (optimizedThreshold, math.pi/2))
-    self.assertTrue(abs(optimizedScore - 1.0) <= 0.00001,
-      "Optimizer returned max value of %r, but expected 1.0" % optimizedScore)
-
-    # Start right of local min
-    optimizedThreshold, optimizedScore = twiddle(objFunction=sine,
-      args = (),
-      initialGuess = math.pi*(1.5 + 0.1),
-      domain = (0, 3*math.pi))
-
-    self.assertTrue(abs(optimizedThreshold - math.pi*5/2) <= 0.00001,
-      "Optimizer returned x = %r, which is not within the tolerance of the \
-      maximum location x = %r" % (optimizedThreshold, math.pi*5/2))
-    self.assertTrue(abs(optimizedScore - 1.0) <= 0.00001,
-      "Optimizer returned max value of %r, but expected 1.0" % optimizedScore)
-
-
-  def testMaxOfGLFunction(self):
-    """Tests limits of the optimizer; not robust enough to find global max 
-    amongst many local minima
-    Optimizer should return 5.0625 +/- tolerance at x=2.5"""
-
-    # Start right of a local min
-    optimizedThreshold, optimizedScore = twiddle(objFunction=GLFunction,
-      args = (),
-      initialGuess = 1,
-      domain = (0.5, 2.5))
-
-    self.assertTrue(abs(optimizedThreshold - 2.5) <= 0.00001,
-      "Optimizer returned x = %r, which is not within the tolerance of the \
-      maximum location x = 2.5" % optimizedThreshold)
-    self.assertTrue(abs(optimizedScore - 5.0625) <= 0.00001,
-      "Optimizer returned a max value of %r, but expected 5.0625"
-      % optimizedScore)
-
-    # Start at a local max
-    optimizedThreshold, optimizedScore = twiddle(objFunction=GLFunction,
-      args = (),
-      initialGuess = 1.25,
-      domain = (0.5, 2.5))
-
-    self.assertFalse(abs(optimizedThreshold - 2.5) <= 0.00001,
-      "Optimizer found the max at the correct x = 2.5 but should not have")
-    self.assertFalse(abs(optimizedScore - 5.0625) <= 0.00001,
-      "Optimizer found the global max value of 5.0625 but should not have")
-
-
-if __name__ == '__main__':
+if __name__ == "__main__":
   unittest.main()
```

Comment:
It already has a newline at the end of the file. Note that a file ending with a newline is different than a file with a blank line at the end.